cloudinit/stages.py

```diff
@@ -431,7 +679,7 @@ def _get_ipath(self, subname=None):
             )
         return instance_dir
 
-    def _write_network_config_json(self, netcfg: dict):
+    def _write_network_config_json(self, netcfg: Optional[dict]):
```

Comment:
Because `self._find_networking_config()` has codepaths that return `None` as the first argument.